src/samples.jl

```diff
@@ -495,3 +495,31 @@ function Base.show(io::IO, samples::Samples)
         show(io, "text/plain", samples.data)
     end
 end
+
+#####
+##### Arrow conversion
+#####
+
+const SamplesArrowType{T,S} = NamedTuple{(:data, :info, :encoded),Tuple{Vector{T},S,Bool}} where {S<:SamplesInfoArrowType}
+
+const SAMPLES_ARROW_NAME = Symbol("JuliaLang.Samples")
+
+Arrow.ArrowTypes.arrowname(::Type{<:Samples}) = SAMPLES_ARROW_NAME
+
+Arrow.ArrowTypes.ArrowType(::Type{<:Samples{D,S}}) where {D,S} = SamplesArrowType{eltype(D),Arrow.ArrowTypes.ArrowType(S)}
+
+function Arrow.ArrowTypes.toarrow(samples::Samples)
+    return (data=vec(samples.data),
+            info=Arrow.ArrowTypes.toarrow(samples.info),
+            encoded=samples.encoded)
+end
+
+function Arrow.ArrowTypes.JuliaType(::Val{SAMPLES_ARROW_NAME}, ::Type{SamplesArrowType{T,S}}) where {T,S}
+    return Samples{Matrix{T},Arrow.ArrowTypes.JuliaType(Val(SAMPLES_INFO_ARROW_NAME), S)}
+end
+
+function Arrow.ArrowTypes.fromarrow(::Type{<:Samples}, arrow_data, arrow_info, arrow_encoded)
+    info = SamplesInfo(arrow_info; validate=false)
```

Comment:
I wonder if `validate=false` is the right choice? I guess we can assume it's properly formulated since usually we will be validating on construction and then serializing/deserializing

Reply:
>  I guess we can assume it's properly formulated since usually we will be validating on construction and then serializing/deserializing

Yeah, this is the assumption we're making